includes/api/class-wc-admin-rest-leaderboards-controller.php

```diff
@@ -49,18 +49,251 @@ public function register_routes() {
 		);
 	}
 
+	/**
+	 * Get the data for the coupons leaderboard.
+	 *
+	 * @param int    $per_page Number of rows.
+	 * @param string $after Items after date.
+	 * @param string $before Items before date.
+	 */
+	public function get_coupons_leaderboard( $per_page, $after, $before ) {
+		$coupons_data_store = new WC_Admin_Reports_Coupons_Data_Store();
+		$coupons_data       = $coupons_data_store->get_data(
+			array(
+				'orderby'       => 'orders_count',
+				'order'         => 'desc',
+				'after'         => $after,
+				'before'        => $before,
+				'per_page'      => $per_page,
+				'extended_info' => true,
+			)
+		);
+
+		$rows = array();
+		foreach ( $coupons_data->data as $coupon ) {
+			$coupon_url  = wc_admin_url( 'analytics/coupons?filter=advanced&coupon_includes=' . $coupon['coupon_id'] );
```

Comment:
Should these links include the date portions of the query as well like the dashboard components? See: https://github.com/woocommerce/wc-admin/blob/fc678aa68583bc7db4df5184d11aa03ca81b7b72/client/dashboard/leaderboards/top-coupons.js#L63

Reply:
I think the lines note here are from an outdated version.  `$persisted_query` is passed to the URL here which will include date params.

Reply:
Ah, I misunderstood that query params had to be duplicated in `persisted_query` to get passed to the URLs. 👍